### PR TITLE
Adds gradient style/colour selection to the appearance changer

### PR DESCRIPTION
**Pull request by Llywelwyn**
now the ghostspawner and augs and stuff can change gradient style/colour too where applicable

image

changes:
  - rscadd: "Added gradient style/colour selection to the appearance changer."
  - bugfix: "Fixed hair colour being unable to be the same as eye colour."

### CODE REVIEW

**Comment by Llywelwyn** [comment]
!review

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 1]
```diff
@@ -48,6 +48,21 @@
 	update_hair()
 	return 1
 
+/mob/living/carbon/human/proc/change_gradient(var/gradient)
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:51]: DMDoc this function

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 2]
```diff
@@ -48,6 +48,21 @@
...
+	if(g_style == gradient)
+		return
+
+	if(!(gradient in GLOB.hair_gradient_styles_list))
+		return
+
+	g_style = gradient
+
+	update_hair()
+	return 1
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:64]: No magic numbers, if it's supposed to be TRUE, use the TRUE define

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 3]
```diff
@@ -106,6 +121,19 @@
 	update_hair()
 	return 1
 
+/mob/living/carbon/human/proc/change_gradient_color(var/red, var/green, var/blue)
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:124]: DMDoc

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 4]
```diff
@@ -106,6 +121,19 @@
...
+		return
+
+	r_grad = red
+	g_grad = green
+	b_grad = blue
+
+	force_update_limbs()
+	update_body()
+	update_hair()
+	return 1
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:135]: No magic numbers as above

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 5]
```diff
@@ -236,6 +264,19 @@
 
 	return valid_hairstyles
 
+/mob/living/carbon/human/proc/generate_valid_gradients()
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:267]: DMdoc

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 6]
```diff
@@ -236,6 +264,19 @@
...
+	var/list/valid_gradient_styles = new()
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/appearance.dm:268]:
```suggestion
	var/list/valid_gradient_styles = list()
```

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**